### PR TITLE
Allow ValidationDependencies to track generic hashable types

### PR DESCRIPTION
**Pull request by maackle**
### Summary

I'm going to need to track EntryHashed dependencies for DPKI validation, so this is groundwork for that.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain/src/conductor/tests/install_app_bundle.rs [hunk 1]
```diff
@@ -424,6 +423,155 @@ async fn network_seed_regression() {
...
+
+    struct TestcaseCommon {
+        conductor: SweetConductor,
+        dnas: Vec<DnaFile>,
+        tmp: TempDir,
+        _start: std::time::Instant,
+    }
+
+    #[derive(Clone, Copy, Debug, strum_macros::ToString)]
+    enum Location {
```
Comment (maackle) [inline crates/holochain/src/conductor/tests/install_app_bundle.rs:436]: I snuck this change in here, it's totally unrelated, just moving a bunch of feature-gated code to live under that feature gate to squelch warnings

File: crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs [hunk 2]
```diff
@@ -1,40 +1,87 @@
...
+    //     fn from((record, fetched_from): (Record, CascadeSource)) -> Self {
+    //         Self {
+    //             dependency: Some(ValidationDependency {
+    //                 signed_action: record.signed_action,
+    //                 fetched_from,
+    //             }),
+    //         }
+    //     }
+    // }
+
```
Comment (maackle) [inline crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs:80]:
```suggestion
```

File: crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs [hunk 3]
```diff
@@ -1,40 +1,87 @@
...
+    collections::{HashMap, HashSet},
+    sync::Arc,
 };
-use std::collections::{HashMap, HashSet};
+
+/// Cache for tracking dependencies which are fetched during sys validation.
+#[derive(Clone)]
+pub struct ValDeps {
+    /// Dependencies found in the same DHT as the dependent
+    pub local: Arc<parking_lot::Mutex<ValidationDependencies<SignedActionHashed>>>,
```
Comment (ThetaSinner) [inline crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs:13]: I wonder if `local` is a bit of an overloaded name to use here? e.g. these dependencies aren't necessarily local, they might be remote and need fetching. Maybe `current_dna`?

**Review by ThetaSinner — APPROVED** [review]

File: crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs (hunk 3 above)
Comment (maackle) [reply crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs:13]: Yes that makes more sense

**Comment by maackle** [comment]
Already included in https://github.com/holochain/holochain/pull/3463

**Pull request closed without merging**